libs/forklift_inventory.py

```
@@ -406,10 +418,116 @@ def __init__(self, client: DynamicClient, provider_name: str, namespace: str) ->
+                sleep=sleep,
+                func=lambda: self.hosts,
+            ):
+                if sample:
+                    LOGGER.info(f"Found {len(sample)} hosts in inventory for provider '{self.provider_name}'")
+                    return sample
+        except TimeoutExpiredError:
+            pass
+
+        raise TimeoutExpiredError(
+            f"No hosts appeared in Forklift inventory for provider '{self.provider_name}' after {timeout}s"
+        )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)